hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java

```diff
@@ -236,38 +223,24 @@ public void run() {
 
                 iter.remove();
 
-                notifyException(invocation, connection, isAlive);
-                Exception ex = newTargetDisconnectedExceptionCausedByHeartbeat(
-                        connection.getRemoteEndpoint(),
-                        connection.toString(),
-                        connection.getLastHeartbeatRequestedMillis(),
-                        connection.getLastHeartbeatReceivedMillis(),
-                        connection.lastReadTimeMillis(),
-                        connection.getCloseCause());
-                invocation.notifyException(ex);
+                notifyException(invocation, connection);
             }
             if (expiredConnections != null) {
                 logExpiredConnections(expiredConnections);
             }
         }
 
-        private void notifyException(ClientInvocation invocation, ClientConnection connection, boolean isAlive) {
```

Comment:
what happens if isAlive changes from true to false while executing this method? I think it is better we pass the parameter in.